fluss-server/src/main/java/org/apache/fluss/server/zk/data/lake/LakeTableSnapshotJsonSerde.java

```diff
@@ -18,168 +18,97 @@
 import org.apache.fluss.metadata.TableBucket;
 import org.apache.fluss.shaded.jackson2.com.fasterxml.jackson.core.JsonGenerator;
 import org.apache.fluss.shaded.jackson2.com.fasterxml.jackson.databind.JsonNode;
 import org.apache.fluss.utils.json.JsonDeserializer;
-import org.apache.fluss.utils.json.JsonSerdeUtils;
 import org.apache.fluss.utils.json.JsonSerializer;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Comparator;
 import java.util.HashMap;
-import java.util.Iterator;
-import java.util.List;
 import java.util.Map;
-import java.util.TreeMap;
-
-import static org.apache.fluss.utils.Preconditions.checkNotNull;
-import static org.apache.fluss.utils.Preconditions.checkState;
 
 /**
  * Json serializer and deserializer for {@link LakeTableSnapshot}.
  *
- * <p>This serde supports two storage format versions:
+ * <p><b>Note:</b> This class is primarily used for backward compatibility to deserialize legacy
+ * version 1 lake snapshot data stored in ZooKeeper. The current storage format (version 2) stores
+ * only file paths in ZooKeeper, with actual snapshot data stored in remote files. This serde is
+ * used by {@link LakeTableJsonSerde} to handle version 1 format deserialization.
+ *
+ * <p>The version 1 format stores the full {@link LakeTableSnapshot} data directly in the ZooKeeper
+ * node, which includes:
  *
  * <ul>
- *   <li>Version 1 (legacy): Each bucket object contains full information including repeated
- *       partition names and partition_id in each bucket entry.
- *   <li>Version 2 (current): Compact format that uses different property keys for partitioned and
- *       non-partitioned tables to simplify deserialization:
- *       <ul>
- *         <li>Non-partition table uses "bucket_offsets": [100, 200, 300], where array index
- *             represents bucket id (0, 1, 2) and value represents log_end_offset. For buckets
- *             without end offset, -1 is written. Missing bucket ids in the sequence are also filled
- *             with -1.
- *         <li>Partition table uses "partition_bucket_offsets": {"1": [100, 200], "2": [300, 400]},
- *             where key is partition id, array index represents bucket id (0, 1) and value
- *             represents log_end_offset. For buckets without end offset, -1 is written. Missing
- *             bucket ids in the sequence are also filled with -1.
- *       </ul>
- *       During deserialization, values of -1 are ignored and not added to the bucket log end offset
- *       map.
+ *   <li>version: 1
+ *   <li>snapshot_id: the snapshot ID
+ *   <li>table_id: the table ID (derived from the first bucket)
+ *   <li>buckets: array of bucket objects, each containing bucket_id, optional partition_id, and
+ *       log_end_offset
  * </ul>
+ *
+ * @see LakeTableJsonSerde for the current format (version 2) that uses this serde for legacy
+ *     compatibility
  */
 public class LakeTableSnapshotJsonSerde
```

Comment:
Consider to rename this to `LakeTableSnapshotLegacyJsonSerde` (Currenlty there is too many `LakeXxxSerde`)